content/en/blog/_posts/scheduler-plugin-reusing/index.md

```diff
@@ -0,0 +1,115 @@
+
+## Alternative approaches for plugin reuse
+
+### Ctrl-C, Ctrl-V
+
+> **Warning**
+> The simplest and probably most frowned upon code-reuse practice is of course copy-paste. Due to well known issues with copy-paste code, this approach can't be recommended, and is mentioned here only as an example of what not to do. Copy-pasting won't be any faster anyway, as you will see below.
+
+### Plugin Composition
+
+Typically a valid option for reusing something is composition. Nothing really stops instantiating an existing plugin in the factory function of a new plugin, and then wrapping things up so that the implementation of the already existing plugin gets reused instead of rewriting the code.
+
+A minimal example of such plugin composite reduces to three main parts. First, one declares the composite plugin interface(s) inside the new plugin's type:
+
+```go
+type CompositionExample struct {
+	framework.ScorePlugin
+}
+```
+
+There we have a placeholder for the ScorePlugin composite in the new plugin's type. Next, it needs to be instantiated in the new plugin's factory function:
+
+```go
+fit, err := noderesources.NewFit(args.CompositeArgs, handle, fts)
+if err != nil {
+    return nil, fmt.Errorf("fit instantiation failed: %w", err)
+}
+
+ce := CompositionExample{
+    ScorePlugin: fit.(framework.ScorePlugin),
+}
+```
+
+Above are the essential parts in order to get NodeResourceFit scores calculated, and it certainly didn't take too many lines of code. Normal plugin boilerplate is required of course as well, and the new plugin needs to be configured as an enabled score-plugin in the scheduler profile.
+
+There are some caveats, as usual. Arguments for the composite are needed, which basically means the args of the new plugin needs to contain and validate the args for the composite also. That may look rather redundant in the scheduler configuration, especially in case the args match exactly what the _NodeResourcesFit_ plugin is doing for individual node scoring, anyhow. The other caveat is, that the new plugin is then running the same node resource calculations another time, which isn't ideal.
+
+Putting these unfortunate drawbacks aside, getting the newly calculated node resource scores back to the new plugin is now only lacking a little bit of code:
+
+```go
+// NormalizeScore implements framework.ScoreExtensions.
+func (ce *CompositionExample) NormalizeScore(ctx context.Context, state *framework.CycleState, p *v1.Pod, scores framework.NodeScoreList) *framework.Status {
+    // logic for using and adjusting the incoming scores goes here
```

Comment:
Indention is inconsistent here (tabs vs. spaces).

Reply:
fixed